Ask user to move to Applications folder on launch

stqism opened the pull request:
Works perfectly, I think it looks neat enough.

![screen shot 2015-03-03 at 7 26 08 pm](https://cloud.githubusercontent.com/assets/2031258/6477689/307f217e-c1db-11e4-93c8-3340772b0621.png)

(Also, I want to drop the dmg for a self-extracting zip because it's nicer to use and takes less work and this is the first step)

stqism committed:
OS X applications folder support

stqism committed:
Almost forgot mv is harmful

Note that this is required because Qt applications cannot run under setuid(0) as qtox_sudo sets

.

Fix quitting

dubslow commented:
The linewrapping looks funny, also period please

dubslow commented:
Did you consider using GUI::askQuestion?

As for your earlier comments about issues with qApp->exit(), it may not be possible to do magic qApp stuff from main... just a thought

zetok commented on src/main.cpp:
```diff
@@ -29,11 +29,16 @@
 
+#include <unistd.h>
+
+#define EXIT_UPDATE_MACX 218 //We track our state using unique exit codes when debugging
```
Where are those codes listed?

stqism commented:
@dubslow AskQuestion isn't real. Also, Qt is pretty dumb with formatting and totally ignores resizing hints so this is by far the best it's going to be. Added the period btw.

Also also also, fixed those issues I was having.

tux3 commented on src/main.cpp:
```diff
@@ -119,6 +124,44 @@ int main(int argc, char *argv[])
+            QString sudo = bindir + "/qtox_sudo rsync -avzh --remove-source-file " + appdir + " /Applications/qtox.app";
+            QString qtox = "open /Applications/qtox.app";
+
+            if (fork() != 0) { //uTox grade cheap hack
```
This comment is not really needed.

tux3 commented on src/main.cpp:
```diff
@@ -119,6 +124,44 @@ int main(int argc, char *argv[])
+        AskInstall.setStandardButtons(QMessageBox::Yes|QMessageBox::No);
+        AskInstall.setDefaultButton(QMessageBox::Yes);
+
+        int AskInstallAttempt = AskInstall.exec(); //Check if user sucks
```
Let's not start insulting our users in the source code, shall we :) ?

stqism committed:
Political correctness

stqism commented:
@tux3 Changed.

tux3 commented on src/main.cpp:
```diff
@@ -119,6 +124,44 @@ int main(int argc, char *argv[])
+
+            QString bindir = qApp->applicationDirPath();
+            QString appdir = bindir;
+            appdir.chop(15);
```
This is really hackish, but I guess it'll have to do for now.
